shopelectro/views/catalog.py

```diff
@@ -178,49 +179,75 @@ def merge_products_context(products):
 @set_csrf_cookie
 class CategoryPage(catalog.CategoryPage):
 
+    def get_products(self) -> QuerySet:
+        sorting_index = int(self.kwargs.get('sorting', 0))
+        sorting_option = SortingOption(index=sorting_index)
+        return models.Product.actives.get_category_descendants(
+            self.object.model, ordering=(sorting_option.directed_field,)
+        )
+
     def get_context_data(self, **kwargs):
         """Add sorting options and view_types in context."""
         context = super().get_context_data(**kwargs)
-        products_on_page = int(self.request.GET.get(
-            'step', get_products_count(self.request),
-        ))
-        page_number = int(self.request.GET.get('page', 1))
-        view_type = self.request.session.get('view_type', 'tile')
-        sorting_index = int(self.kwargs.get('sorting', 0))
-        sorting_option = SortingOption(index=sorting_index)
-        category = context['category']
-        if (
-            page_number < 1 or
-            products_on_page not in settings.CATEGORY_STEP_MULTIPLIERS
-        ):
-            raise http.Http404('Page does not exist.')  # Ignore CPDBear
 
-        all_products = models.Product.actives.get_category_descendants(
-            category, ordering=(sorting_option.directed_field, )
-        )
+        products = self.get_products()
 
         group_tags_pairs = (
             models.Tag.objects
-            .filter_by_products(all_products)
+            .filter_by_products(products)
             .get_group_tags_pairs()
         )
 
-        tags = self.kwargs.get('tags')
+        return {
+            **context,
+            'products_data': merge_products_context(products),
+            'group_tags_pairs': group_tags_pairs,
+        }
 
-        tag_titles = ''
-        if tags:
-            slugs = models.Tag.parse_url_tags(tags)
-            tags = models.Tag.objects.filter(slug__in=slugs)
 
-            all_products = (
-                all_products
+# @todo #550:60m Split TaggedCategoryPage class
+#  See details at parent task.
+class TaggedCategoryPage(CategoryPage):
+
+    def get_sorting_index(self):
```

Comment:
Seems the better place for it in `CategoryPage` class

Reply:
@artemiy312 , the best place is SortedCategoryPage.
I moved all instead of CategoryPage's logic to TaggedCategoryPage.
Then i'll fork another class from TaggedCategoryPage and so on